src/components/overlay/README.md

```diff
@@ -0,0 +1,167 @@
+# UiOverlay
+Overlay system that handles multiple instances, transitions, scrolling behaviour and more.
+
+## Usage
+
+0. Add the component to the location where you want to insert one or multiple overlays.
+You can also choose to add multiple instances of this component in different locations, each displaying only a subset.
+ 
+```vue
+<script>
+    import { mapState } from 'vuex';
+    import { UiOverlay } from '@dreipol/vue-ui/src/components';
+    
+    export default {
+        components: {
+            UiOverlay,
+        },
+        computed: {
+            ...mapState('vue-ui/overlay', ['overlays']),
+        },
+    };
+</script>
+
+<template>
+    <div>
+        <ui-overlay v-for="(value, key) in overlays" :id="key" :key="key"/>
+    </div>
+</template>
+```
+
+0. Add the vuex overlay module to your own store.
+
+```js
+import Vuex from 'vuex';
+import overlay from '@dreipol/vue-ui/src/vuex/modules/overlay';
+
+const store = new Vuex.Store({
+    modules: {
+        'vue-ui': {
+            namespaced: true,
+            modules: { overlay },
+        },
+    },
+});
+```
+
+0. Use the overlay vuex module's actions to open and close your overlays. 
+You can use them in your components or even in other vuex actions.
+
+```vue
+<template>
+    <button @click="onClick">
+        click me to open the overlay                
+    </button>
+</template>
+
+<script>
+    export default {
+        methods: {
+            ...mapActions('vue-ui/overlay', ['openOverlay', 'closeOverlay']),
+            onClick() {
+                this.openOverlay({
+                    id: 'MY_OVERLAY_ID',
+                    component: 'my-overlay-content-comopnent',
+                });
+            },
+        },
+    };
+</script>
+```
+
+
+## Props
```

Comment:
I would rather call it config. Props in the vue context stand for something different. 

I also would add an example so the user understands which parameter has to follow which interface. 
Like 

```js
onClick() {
this.openOverlay(config: IOverlay);
},
```

Reply:
Right, that was copy waste